### PR TITLE
👩‍🔬 Inline citation improvements

### PR DESCRIPTION
**Pull request by rowanc1**
- Allow macros in tex parsing
- Prefix and suffix for inline citations

cite-prefix-suffix

### CODE REVIEW

File: packages/tex-to-myst/src/citations.ts [hunk 1]
```diff
@@ -34,6 +25,16 @@ function createCitation(
   } else {
     cite.label = citations[0];
   }
+  const args = getArguments(node, 'argument');
+  if (args.length > 0) {
+    // If there is only one argument, it is the suffix
+    const [prefix, suffix] = args.length === 1 ? [undefined, args[0]] : args;
+    const prefixText = texToText(prefix).trim();
+    const suffixText = texToText(suffix).trim();
+    const citation = isGroup ? cite.children?.[0] : cite;
```
Comment (fwkoch) [inline packages/tex-to-myst/src/citations.ts:34]: This feels a little funny if there are multiple citations and prefix/suffix, but maybe this is just the expected behaviour and not a problem.

**Review by fwkoch — APPROVED** [review]
This looks nice! Clean and pretty straightforward, prefix/suffix on `Cite` makes sense to me.

File: packages/tex-to-myst/src/citations.ts (hunk 1 above)
Comment (rowanc1) [reply packages/tex-to-myst/src/citations.ts:34]: Yep, you are correct. This should be on the first and last child of the group:

image

image

Will fix that up.

Comment (rowanc1) [reply packages/tex-to-myst/src/citations.ts]: Ok, pushed. I think there are actually quite a few intricacies when you get into textual/narrative citations. Where the prefix is repeated, but not the suffix. I am going to leave as is for now, and fix when we get complaints. Lots of weird work-arounds on the internet for this type of stuff.

image

image

**Pull request merged**